src/plugins/invitationmanager/gcim.js

```diff
@@ -0,0 +1,720 @@
+		// those ones are removed and not tested on future pages in the visit
+
+		var surveySubList = {};
+		var url = document.location.hostname + document.location.pathname;
+		var count = 0;
+
+		for ( var i = 0; i < surveyDB.surveys.length; i++ ) {
+			var survey_i = surveyDB.surveys[ i ];
+
+			if ( overrideID ) {
+
+				// Skip this loop's logic if it's not the desired survey,
+				// therefore there's no chance it will be selected
+				if ( survey_i.id !== overrideID ) {
+					continue;
+				}
+
+				// If it is the desired survey, set it's probability to 100.
+				// Note that the desired survey must still be applicable on this page
+				// so the logic in the swtich statement still must execute meet the conditions
+				survey_i.probability = 100;
+			}
+
+			switch ( survey_i.type ) {
+
+			case "Entire site" :
+
+				// Get an object array of the sites to run the survey on
+				var sublist = getObjArr( survey_i[ "survey-urls" ] );
+
+				for ( var c = 0; c < sublist.length; c++ ) {
+
+					//convert the value for the site into RegExp non-case sensitive
+					var b = new RegExp( sublist[ c ].site, "i" );
+
+					//if the url matches the regEx
+					if ( b.test( url ) ) {
+
+						// Store this survey's id and probability in the new object
+						// (this object is later fed to getWeightedRandom)
+
+						// Convert site survey probability into rate since it is
+						// between 0 and Visitor Allocation for site wide type
+
+						surveySubList[ survey_i.id ] = survey_i.probability * surveyDB.settings[ "site-probability" ];
+
+
+						// Keep track of the number of surveys that apply so we can
+						// console log it later
+						count++;
+
+						// Remove this survey from the DB, since we won't test for it on future pages
+						surveyDB.surveys.splice( i, 1 );
+
+						// Decrement i since the next record has moved to replace the current record
+						i--;
+					}
+				}
+
+				break;
+
+			case "Page" :
+
+				var sublist = getObjArr( survey_i[ "survey-urls" ] );
+				var wLocation = window.location.href.toLowerCase();
+
+				for ( var c = 0; c < sublist.length; c++ ) {
+
+					if ( wLocation.indexOf( sublist[ c ].url.toLowerCase( ) ) !== -1 && i >= 0 ) {
+						surveySubList[ survey_i.id ] = survey_i.probability * surveyDB.settings[ "page-probability" ];
+						count++;
+
+						// Remove the eligible survey from the DB, since we won't test for it on future pages
+						surveyDB.surveys.splice( i, 1 );
+
+						// Decrement i since the next record has moved to replace the current record
+						i--;
+					}
+				}
+
+				break;
+			}
+		}
+
+		consoleLog( count + " potential surveys, selecting a survey" );
+
+		return surveySubList;
+	},
+
+
+	/*
+	* Display the popup given the survey parameters
+	*/
+	invite = function( survey ) {
+
+		var html =
+		"<aside id='gc-im-popup' class='wb-overlay wb-overlay-im modal-content-im overlay-def wb-popup-im ' tabindex='0'>" +
```

Comment:
@djamilab Add 1 more tab indentation. 

@GormFrank eslint usually grab that kind of error, we might require a config change in our build script